src/collection/List.scala

```diff
@@ -105,10 +105,17 @@ abstract sealed class List[+A] {
   /**
    * Generates all the prefixes of this list.
    * 
-   * Time - O(n)
+   * Time - O(n^2)
    * Space - O(n)
    */
-  def prefixes: List[List[A]] = ???
+  def prefixes: List[List[A]] = {
+    def helper(acc: List[List[A]], r: List[A]) : List[List[A]] = {
```

Comment:
This is usually called `loop` but this is not a big deal so we can keep the name.